src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1461,21 +1463,44 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         $container->setAlias('messenger.transport.encoder', $config['encoder']);
         $container->setAlias('messenger.transport.decoder', $config['decoder']);
 
+        if (null === $config['default_bus']) {
+            if (count($config['buses']) > 1) {
+                throw new LogicException(sprintf('You need to define a default bus with the "default_bus" configuration. Possible values: %s', implode(', ', array_keys($config['buses']))));
+            }
+
+            $config['default_bus'] = array_keys($config['buses'])[0];
+        }
+
+        $defaultMiddlewares = array('before' => array('logging'), 'after' => array('route_messages', 'call_message_handler'));
+        foreach ($config['buses'] as $name => $bus) {
+            $busId = 'messenger.bus.'.$name;
+
+            $middlewares = $bus['default_middlewares'] ? array_merge($defaultMiddlewares['before'], $bus['middlewares'], $defaultMiddlewares['after']) : $bus['middlewares'];
+            $container->setParameter($busId.'.middlewares', $middlewares);
+            $container->setDefinition($busId, (new Definition(MessageBus::class, array(array())))->addTag('messenger.bus', array('name' => $name)));
+
+            if ($name === $config['default_bus']) {
+                $container->setAlias('message_bus', $busId);
+                $container->setAlias(MessageBusInterface::class, $busId);
+            }
+
+            $container->setDefinition(
```

Comment:
should/can we process this at the compile pass level?

Reply:
Yes, very good idea! So we add it only when we need it 💃